### PR TITLE
categorizable objectives

### PR DESCRIPTION
**Pull request by stopfstedt**
companion ticket to ilios/ilios#2854

### CODE REVIEW

File: addon-test-support/ilios-common/page-objects/components/objective-list-item-terms.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { clickable, create, collection, isHidden, isPresent } from 'ember-cli-page-object';
+import selectedTerms from './detail-terms-list';
+
+const definition = {
+  scope: '[data-test-objective-list-item-terms]',
+  list: collection('[data-test-detail-terms-list]', selectedTerms),
+  empty: isHidden('[data-test-parent]'),
```
Comment (jrjohnson) [inline addon-test-support/ilios-common/page-objects/components/objective-list-item-terms.js:7]: Doesn't seem to be an element on this component.

File: addon/components/session-publicationcheck.hbs [hunk 2]
```diff
@@ -65,9 +65,9 @@
                 {{t "general.no"}}
               </td>
             {{/if}}
-            {{#if @session.objectives.length}}
+            {{#if (await @session.objectives)}}
```
Comment (jrjohnson) [inline addon/components/session-publicationcheck.hbs:68]: `has-many-length` here to make this sync.

File: addon/components/session/manage-objective-parents.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { restartableTask } from 'ember-concurrency-decorators';
+import { all } from 'rsvp';
+
+export default class SessionManageObjectiveParents extends Component {
+
+  @tracked courseObjectives = [];
```
Comment (jrjohnson) [inline addon/components/session/manage-objective-parents.js:8]: Confusing property name, maybe just `objectives` or `courseObjectiveObjectives`?

File: config/api-version.js [hunk 4]
```diff
@@ -1,3 +1,3 @@
 /* eslint-env node */
 
-module.exports = 'v1.42';
+module.exports = 'v1.43';
```
Comment (jrjohnson) [inline config/api-version.js:3]: `v2.0.0`

Comment (stopfstedt) [reply config/api-version.js]: fixed.

File: addon/components/session/manage-objective-parents.js (hunk 3 above)
Comment (stopfstedt) [reply addon/components/session/manage-objective-parents.js]: renamed.

File: addon-test-support/ilios-common/page-objects/components/objective-list-item-terms.js (hunk 1 above)
Comment (stopfstedt) [reply addon-test-support/ilios-common/page-objects/components/objective-list-item-terms.js]: yanked.

File: addon/components/detail-terms-list.js [hunk 5]
```diff
@@ -3,11 +3,19 @@ import { tracked } from '@glimmer/tracking';
 import { dropTask } from 'ember-concurrency-decorators';
 import { all } from 'rsvp';
 
-export default class DetailTermsListComponent extends Component {
+export default class DetailTermsList extends Component {
```
[the remark this answers is not in the document]
Comment (stopfstedt) [reply addon/components/detail-terms-list.js]: added back in.

File: addon/components/session-publicationcheck.hbs (hunk 2 above)
Comment (stopfstedt) [reply addon/components/session-publicationcheck.hbs]: convered.

File: addon/components/detail-terms-list.hbs [hunk 6]
```diff
@@ -1,13 +1,19 @@
...
-  {{did-update (perform this.load) this.filteredTerms}}
+  {{did-insert (perform this.load) @terms}}
+  {{did-update (perform this.load) @terms}}
+  data-test-detail-terms-list
 >
   {{#if (is-array this.sortedTerms)}}
     <div data-test-title>
-      <strong>
+      {{#if @manage}}
+      <strong class="link-button" role="button" {{on "click" (fn @manage @vocabulary)}}>
```
[the remark this answers is not in the document]
Comment (stopfstedt) [reply addon/components/detail-terms-list.hbs]: buttonized.

File: addon/components/course-publicationcheck.hbs [hunk 7]
```diff
@@ -50,9 +50,9 @@
               {{t "general.no"}}
             </td>
           {{/if}}
-          {{#if @course.objectives.length}}
+          {{#if (await @course.objectives)}}
             <td class="yes">
-              {{t "general.yes"}} ({{@course.objectives.length}})
+              {{t "general.yes"}} ({{get (await @course.objectives) "length"}})
```
[the remark this answers is not in the document]
Comment (stopfstedt) [reply addon/components/course-publicationcheck.hbs]: thanks. converted.

File: addon/components/course-publicationcheck.hbs [hunk 8]
```diff
@@ -50,9 +50,9 @@
               {{t "general.no"}}
             </td>
           {{/if}}
-          {{#if @course.objectives.length}}
+          {{#if (await @course.objectives)}}
```
[the remark this answers is not in the document]
Comment (stopfstedt) [reply addon/components/course-publicationcheck.hbs]: converted.

**Comment by stopfstedt** [comment]
> Couple of additional notes:
> 
> 1. Need to pre-load course / session / program year objectives and all school terms and vocabularies with other course data in https://github.com/ilios/common/blob/master/addon/utils/preload-course.js

thanks. handled. 
> 2. Not super clear where to click to edit in an objective. Maybe needs a button? Or make the whole box clickable?  Maybe when hovering any of the boxes you get an edit icon in the top right corner? Mayabe it's fine to just click the vocabulary title?

i think it's fine. everything on that row that's blue is click-to-edit, should be intuitive enough.

> 3. Seems to be a missing acceptance test for managing course / session objective terms

coming right up, sit tight.

**Pull request merged**